File: src/pewmode/Flames.java
package pewmode;

import org.newdawn.slick.Image;
import org.newdawn.slick.SlickException;

import java.util.ArrayList;
import java.util.List;

public class Flames extends Sprite {
    List<Spritable> flames = new ArrayList<>();

    Flames(Ship ship) throws SlickException {
        Image rotated = new Image("flame.png");
        rotated.setRotation(90);
        flames.add(new Flame(rotated, ship, 0));

        flames.add(new Flame(new Image("flame.png"), ship, 190));
        flames.add(new Flame(new Image("flame.png"), ship, 210));
        flames.add(new Flame(new Image("flame.png"), ship, 230));
        flames.add(new Flame(new Image("flame.png"), ship, 300));
        flames.add(new Flame(new Image("flame.png"), ship, 320));
        flames.add(new Flame(new Image("flame.png"), ship, 340));

        flames.add(new Flame(new Image("flame.png"), ship, 247));
        flames.add(new Flame(new Image("flame.png"), ship, 267));
        flames.add(new Flame(new Image("flame.png"), ship, 287));
        flames.add(new Flame(new Image("flame.png"), ship, 357));
        flames.add(new Flame(new Image("flame.png"), ship, 377));
        flames.add(new Flame(new Image("flame.png"), ship, 397));
    }

    public void render() {
        if (!menu) {
            for (int i = 0; i < 7; i++) {
                flames.get(i).render();
            }
        } else {
            for (int i = 7; i < 13; i++) {
                flames.get(i).render();
            }
        }
    }

    public void update() {
        for (Spritable s : flames) {
            s.update();
        }
    }
}

File: src/pewmode/Flames.kt
package pewmode

import org.newdawn.slick.Image
import org.newdawn.slick.SlickException

import java.util.ArrayList

class Flames @Throws(SlickException::class)
internal constructor(ship: Ship) : SpriteGroup() {
    internal var flames: MutableList<Sprite> = ArrayList()

    init {
        val rotated = Image("flame.png")
        rotated.rotation = 90f
        flames.add(Flame(rotated, ship, 0))

        flames.add(Flame(Image("flame.png"), ship, 190))
        flames.add(Flame(Image("flame.png"), ship, 210))
        flames.add(Flame(Image("flame.png"), ship, 230))
        flames.add(Flame(Image("flame.png"), ship, 300))
        flames.add(Flame(Image("flame.png"), ship, 320))
        flames.add(Flame(Image("flame.png"), ship, 340))

        flames.add(Flame(Image("flame.png"), ship, 247))
        flames.add(Flame(Image("flame.png"), ship, 267))
        flames.add(Flame(Image("flame.png"), ship, 287))
        flames.add(Flame(Image("flame.png"), ship, 357))
        flames.add(Flame(Image("flame.png"), ship, 377))
        flames.add(Flame(Image("flame.png"), ship, 397))
    }

    override fun render() {
        if (!isMenuMode()) {
            for (i in 0..6) {
                flames[i].render()
            }
        } else {
            for (i in 7..12) {
                flames[i].render()
            }
        }
    }

    override fun update() {
        flames.forEach { it.update() }
    }
}
